openquake/commonlib/oqvalidation.py

```diff
@@ -618,10 +618,18 @@ def is_valid_inputs(self):
         Invalid calculation_mode="{calculation_mode}" or missing
         fragility_file/vulnerability_file in the .ini file.
         """
+        if self.hazard_calculation_id:
+            parent = list(datastore.read(self.hazard_calculation_id))
```

Comment:
Parent can be of different types. I suggest to use only one type for it

Reply:
And you might change `parent` into `parent_datasets`, so it becomes clear that it can contain multiple elements